File: app/src/main/java/com/ed/redditapp/lib/api/standard/StandardSubreddit.java
package com.ed.redditapp.lib.api.standard;

import com.ed.redditapp.lib.api.SubReddit;

import org.json.JSONException;
import org.json.JSONObject;

import lombok.Getter;

@Getter
public class StandardSubreddit implements SubReddit {
    private final String name;
    private final String title;
    private final String description;
    private final String iconUrl;
    private final int subsCount;

    public StandardSubreddit(JSONObject json) throws JSONException {
        name = json.getString("display_name");
        subsCount = json.getInt("subscribers");
        title = json.getString("title");
        description = json.getString("public_description");

        String icon = json.getString("icon_img");
        String communityIcon = json.getString("community_icon");
        iconUrl = !icon.isEmpty() ? icon : communityIcon;
    }
}

File: app/src/main/java/com/ed/redditapp/lib/api/standard/StandardSubreddit.kt
package com.ed.redditapp.lib.api.standard

import com.ed.redditapp.lib.api.SubReddit
import org.json.JSONObject

class StandardSubreddit(json: JSONObject) : SubReddit {
    override val name = json.getString("display_name")
    override val title = json.getString("title")
    override val description = json.getString("public_description")
    override val iconUrl: String
    override val subsCount = json.getInt("subscribers")

    init {
        val icon = json.getString("icon_img")
        val communityIcon = json.getString("community_icon")
        iconUrl = if (icon.isNotEmpty()) icon else communityIcon
    }
}